find_disappeared_numbers.cpp

```diff
@@ -0,0 +1,33 @@
+// Time Complexity : O(n)
+// Space Complexity :  O(n)
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this - Couldn't find the O(1) space solution
```

Comment:
Please try to use the fact that the array has numbers only between 1 and n to optimize on space.